src/rust/zh.dmzj/src/lib.rs

```diff
@@ -0,0 +1,414 @@
+
+            chapters.push(Chapter {
+                id: format!("{}/{}", id.clone(), data.get("id").as_string()?.read()),
+                title: data.get("chapter_name").as_string()?.read(),
+                volume: -1.0,
+                chapter: (data.get("chapter_order").as_int()? / 10) as f32,
+                date_updated: data.get("updatetime").as_int()? as f64,
+                scanlator: String::new(),
+                url: String::new(),
+                lang: String::from("zh"),
+            });
+        }
+    }
+    return Ok(chapters);
+}
+
+#[get_page_list]
+fn get_page_list(id: String) -> Result<Vec<Page>> {
+    // Not Tested
+    // Maybe only use the first one.
+
+    let url = [
+        format!("{}/{}.html", API_PAGELIST_WEBVIEW_URL, &id),
+        format!(
+            "{}/chapter/{}.json?channel=android&version=3.0.0&timestamp={}",
+            V3_API_CHAPTER_URL,
+            &id,
+            aidoku::std::current_date() as i64
+        ),
+        format!("{}/comic/chapter/{}.html", API_PAGELIST_OLD_URL, &id),
+    ];
+    let mut index = 0;
+
+    let arr = loop {
+        if index > 2 {
+            break ArrayRef::new();
+        }
+
+        let req = helper::GET(url[index].clone());
+
+        let req = req.json();
+        let r = match index {
+            0 | 1 => req.as_object()?.get("page_url").clone().as_array().ok(),
+            2 => req
+                .as_object()?
+                .get("chapter")
+                .as_object()?
+                .get("page_url")
+                .clone()
+                .as_array()
+                .ok(),
+            _ => None,
+        };
+        match r {
+            Some(r) => break r,
+            _ => index = index + 1,
+        };
+    };
+
+    let mut pages = Vec::new();
+
+    let mut index = 0;
+
+    for r in arr {
+        let mut imageUrl = r.as_string()?.read();
+        imageUrl = imageUrl
+            .replace("http:", "https:")
+            .replace("dmzj1.com", "dmzj.com");
+
+        let thumbUrl = {
+            if id != "" {
+                let initial = imageUrl
+                    .strip_prefix("https://images.dmzj.com/")
+                    .unwrap()
+                    .get(0..1)
+                    .unwrap();
+
+                format!("{}/{}/{}/{}.jpg", IMAGE_SMALL_URL, initial, id, index)
+            } else {
+                String::new()
+            }
+        };
+
+        pages.push(Page {
+            index,
+            url: helper::encodeURI(&imageUrl),
+            base64: String::new(),
+            text: String::new(),
+        });
+
+        index += 1;
+    }
+
+    Ok(pages)
+}
+
+// Doesn't work
+#[modify_image_request]
+fn modify_image_request(request: Request) {
+    request.header("Referer", "https://www.dmzj.com/");
+}
+
+#[handle_url]
+pub fn handle_url(url: String) -> Result<DeepLink> {
+    Err(aidoku::error::AidokuError {
+        reason: aidoku::error::AidokuErrorKind::Unimplemented,
+    })
+}
```

Comment:
You should implement deep linking for manga links at least.

Reply:
how is this working? just want to test it, but don't know how to use this function in app.

Reply:
Go to a manga or chapter link in safari and then change the url scheme to `aidoku://`. Aidoku should be able to detect the url is for your source and send it to handle_url for you to return the target manga.